### PR TITLE
Fix/sources loading

### PR DESCRIPTION
**Pull request by cilasmarques**
This PR make 2 things:
1 - Fix the source loading. It, waits the source loading to append the layer, avoiding load errors
2 - Load all the sources asynchronously.

To test, access the map page.

### CODE REVIEW

File: src/components/MapTiff/MapTiff.tsx [hunk 1]
```diff
@@ -25,148 +24,143 @@ const MapTiff = ({ isReduced = false, ...props }: { isReduced?: boolean }) => {
...
   );
 
-  const loadMapLayer = useCallback(
-    async (id: string, year: string) => {
-      setLoading(true);
-      await loadSource(id, year);
+  const loadInitialSources = useCallback(
+    async (map: maplibregl.Map) => {
+      // === Add current raster source
+      const yearStr = year ? year : "general";
```
Comment (RodrigoEC) [inline src/components/MapTiff/MapTiff.tsx:94]:
```suggestion
      const yearStr = year || "general";
```

File: src/components/MapTiff/MapTiff.tsx [hunk 2]
```diff
@@ -232,29 +226,61 @@ const MapTiff = ({ isReduced = false, ...props }: { isReduced?: boolean }) => {
...
   useEffect(() => {
-    if (map && id && year) {
-      if (!isReduced) addPopupEffect(id, year);
-      loadMapLayer(id, year);
+    const yearStr = year ? year : "general";
+    const sourceKey = `@oca/${id}${yearStr}`;
 
-      return () => {
+    if (loadedSources.has(sourceKey)) {
+      if (map && !map.getLayer(sourceKey)) {
```
Comment (RodrigoEC) [inline src/components/MapTiff/MapTiff.tsx]:
```suggestion
    if (loadedSources.has(sourceKey) && map && !map.getLayer(sourceKey)) {
```

**Review by RodrigoEC — CHANGES_REQUESTED** [review]
Really good!

Just commented some things on code but there's some other that I saw on the code and I think we'll could improve. I'm gonna list them here:

## Map initialization

image

Could be just like this:

```js
useEffect(() => {
    initializeMap();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```

## Unused if statement
The if statement on AddPopupEffect is never called beacuase the id + year will never be an unkown key

image

File: src/components/MapTiff/MapTiff.tsx [hunk 3]
```diff
@@ -232,35 +223,64 @@ const MapTiff = ({ isReduced = false, ...props }: { isReduced?: boolean }) => {
...
+        minZoom: isReduced ? 3 : 3.6,
+        maxZoom: 10,
+      });
 
-  useEffect(() => {
-    if (map && id && year) {
-      if (!isReduced) addPopupEffect(id, year);
-      loadMapLayer(id, year);
+      newMap.on("load", async () => {
+        if (!isReduced)
```
Comment (RodrigoEC) [inline src/components/MapTiff/MapTiff.tsx:242]: With the new change you actually removed the property of not adding the popup effect on the  MapTiff's reduced version, I think just adding it again could solve the issue

**Review by RodrigoEC — CHANGES_REQUESTED** [review]

**Review by RodrigoEC — APPROVED** [review]
## Great fix!

abbot

**Review by LazaroQueiroz — APPROVED** [review]
LGTM!

**Review by abraaovalentim — APPROVED** [review]
LGTM!

**Pull request merged**